packages/ngtools/webpack/src/angular_compiler_plugin.ts

```diff
@@ -911,18 +911,18 @@ export class AngularCompilerPlugin {
         // When Ivy is enabled we need to add the fields added by NGCC
         // to take precedence over the provided mainFields.
         // NGCC adds fields in package.json suffixed with '_ivy_ngcc'
-        // Example: module -> module__ivy_ngcc
+        // Example: module -> module_ivy_ngcc
         // tslint:disable-next-line:no-any
         (compiler as any).resolverFactory.hooks.resolveOptions
           .for('normal')
           // tslint:disable-next-line:no-any
           .tap('WebpackOptionsApply', (resolveOptions: any) => {
-            const mainFields = (resolveOptions.mainFields as string[])
-              .map(f => [`${f}_ivy_ngcc`, f]);
+            const originalMainFields: string[] = resolveOptions.mainFields;
+            const ivyMainFields = originalMainFields.map(f => `${f}_ivy_ngcc`);
```

Comment:
I am not convinced with this change, this might have the side effect that when not an entrypoint is not processed by NGCC for a reason or another a wrong entrypoint will get added in the compilation.

Example, the UMD bundle was successfully converted with NGCC but the FESM5 wasn't so we'll end up with the 
following;
```
module_ivy_ngcc, main_ivy_ngcc, module, main
```

In this case since module was not converted, the UMD Ivy version of the bundle will be included instead which is not desired and expected and user's will only notice this if they due due to increase in bundle size.

Reply:
Why do you think it would be better to include the unprocessed FESM5 bundle?

Reply:
Cause it will error during run time. So at least the users will know that something went wrong.

Reply:
But this will usually mean that the entry-point was explicitly ignored in ngcc, so it is expected that the entry-point was not converted. See https://github.com/angular/angular-cli/pull/17441#issuecomment-612098944.